### PR TITLE
ObjectLoader: Fix loading multiple materials with the same uuid

### PR DESCRIPTION
**Pull request by mixtur**
pr for https://github.com/mrdoob/three.js/issues/15152

### CODE REVIEW

File: src/loaders/ObjectLoader.js [hunk 1]
```diff
@@ -451,9 +451,10 @@ Object.assign( ObjectLoader.prototype, {
 
 	},
 
-	parseMaterials: function ( json, textures ) {
+    	parseMaterials: function ( json, textures ) {
```
Comment (Mugen87) [inline src/loaders/ObjectLoader.js]: Can you please correct the formatting in this line?

**Review by Mugen87 — CHANGES_REQUESTED** [review]

**Comment by Mugen87** [comment]
Apart from the formatting issue, I think the changes look good so far.

File: src/loaders/ObjectLoader.js (hunk 1 above)
Comment (mixtur) [reply src/loaders/ObjectLoader.js]: done

**Comment by WestLangley** [comment]
>Apart from the formatting issue, I think the changes look good so far.

Thanks for testing this. Stating _exactly_ what you approve is preferable to clicking the approve button. :-)

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

File: src/loaders/ObjectLoader.js [hunk 2]
```diff
@@ -466,21 +467,31 @@ Object.assign( ObjectLoader.prototype, {
...
+							var mat = loader.parse( data.materials[ j ] );
+							cache[ data.materials[ j ].uuid ] = mat;
+							array.push( mat );
+
+						}
 
 					}
 
 					materials[ data.uuid ] = array;
+					cache[ data.uuid ] = array;
```
Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: Is this one needed?

File: src/loaders/ObjectLoader.js [hunk 3]
```diff
@@ -466,21 +467,31 @@ Object.assign( ObjectLoader.prototype, {
...
 
 				} else {
 
 					materials[ data.uuid ] = loader.parse( data );
+					cache[ data.uuid ] = materials[ data.uuid ];
```
Comment (mrdoob) [inline src/loaders/ObjectLoader.js]: This one too. Is it really needed?

Comment (mixtur) [reply src/loaders/ObjectLoader.js]: this is most likely needed.
For cases `MultiMaterial = [X, Y], AnotherMaterial = X`

File: src/loaders/ObjectLoader.js (hunk 2 above)
Comment (mixtur) [reply src/loaders/ObjectLoader.js]: Maybe not

File: src/loaders/ObjectLoader.js (hunk 3 above)
Comment (mrdoob) [reply src/loaders/ObjectLoader.js]: I see...

File: src/loaders/ObjectLoader.js (hunk 2 above)
Comment (mrdoob) [reply src/loaders/ObjectLoader.js]: Would you like to do another PR removing it? Just to keep the code clear.

Comment (mixtur) [reply src/loaders/ObjectLoader.js]: sure

Comment (mrdoob) [reply src/loaders/ObjectLoader.js]: Actually, I'll clean it up on my end 👌

Comment (mrdoob) [reply src/loaders/ObjectLoader.js]: https://github.com/mrdoob/three.js/commit/f8dc0a9335a18eda38bf028e54cca562ea11e0d5

Comment (mrdoob) [reply src/loaders/ObjectLoader.js]: screen shot 2018-10-31 at 10 39 05 am

**Comment by mixtur** [comment]
@mrdoob There is still a problem https://github.com/mrdoob/three.js/blob/dev/src/loaders/ObjectLoader.js#L500.
Single materials are never taken from cache, and always being parsed. So it still can be duplicated if MultiMaterial was parsed before single material.
Should I do another PR?

**Pull request merged**